### ISSUE TITLE
Failed to parse Redshift SIMILAR TO regexp

### ISSUE THREAD

**Issue by Densol92** [sqlfluff/sqlfluff#5445]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Failed  to parse valid redshift regexp

### Expected Behaviour

No parse errors

### Observed Behaviour

L:   1 | P:  57 |  PRS | Line 1, Position 57: Found unparsable section: "]+'"

### How to reproduce

1. create sql file test.sql

```
SELECT '1' SIMILAR TO '[][}{|~^ _`@+!":#>*<$\\,/?%&\'();]+'
```
2. run command
`sqlfluff lint -d redshift --ignore-local-config test.sql`

### Dialect

redshift

### Version

sqlfluff, version 2.3.5

### Configuration

reproduced on default configuration

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by pilou-komoot** [comment]
Hi @Densol92,
The query you submitted to reproduce does not follow SQL standards. When you want to escape a single quote, you have to double it, like `''`.

The following code works, and is properly parsed by SQLFluff:
```sql
SELECT '1' SIMILAR TO '[][}{|~^ _`@+!":#>*<$\\,/?%&''();]+';

-- instead of

SELECT '1' SIMILAR TO '[][}{|~^ _`@+!":#>*<$\\,/?%&\'();]+';
```

This issue can be closed.
